File: packages/coterie-js/src/state.ts
export interface AgentRun {
  agent_id: string;
  prompt: string;
  stdout: string;
  stderr: string;
  exit_code: number;
  files_changed: string[];
  duration_s: number;
  cost_estimate_usd: number | null;
}

export type Status =
  | "planning"
  | "routing"
  | "executing"
  | "judging"
  | "awaiting_human"
  | "done"
  | "failed";

export interface CoterieState {
  task: string;
  plan: string[];
  current_step_idx: number;
  runs: AgentRun[];
  artifacts: Record<string, string>;
  last_winner: string | null;
  status: Status;
  config: Record<string, unknown>;
  spend_usd: number;
}

File: packages/coterie-py/src/coterie/state.py
from operator import add
from typing import Annotated, Any, Literal, TypedDict


class AgentRun(TypedDict):
    agent_id: str
    prompt: str
    stdout: str
    stderr: str
    exit_code: int
    files_changed: list[str]
    duration_s: float
    cost_estimate_usd: float | None


Status = Literal[
    "planning",
    "routing",
    "executing",
    "judging",
    "awaiting_human",
    "done",
    "failed",
]


class CoterieState(TypedDict):
    task: str
    plan: list[str]
    current_step_idx: int
    runs: Annotated[list[AgentRun], add]
    artifacts: dict[str, str]
    last_winner: str | None
    status: Status
    config: dict[str, Any]
    spend_usd: float
